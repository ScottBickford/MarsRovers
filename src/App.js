import React from "react";
import "./App.css";
import { Container } from "semantic-ui-react";
import RoverInstructionsPage from "./features/roverInstructions/RoverInstructionsPage";

function App() {
  return (
    <Container className='main'>
      <h2>Rover Instructions</h2>
      <RoverInstructionsPage />
    </Container>
  );
}

export default App;
