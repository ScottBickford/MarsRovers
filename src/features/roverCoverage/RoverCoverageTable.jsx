import React from 'react';
import { Table } from 'semantic-ui-react';

export const RoverCoverageTable = ({ upperRightCoords, positions }) => {
  const createTable = () => {
    let table = [];
    for (let y = 0; y <= upperRightCoords.y; y++) {
      let children = [];
      for (let x = 0; x <= upperRightCoords.x; x++) {
        const markSquare = positions.some(
          p => p.x === x && upperRightCoords.y - p.y === y // The table is inverted on the y axis, which is why we use upperRightCoords.y - p.y 
        );
        children.push(
          <Table.Cell
            key={x}
            style={{ backgroundColor: markSquare ? 'lightblue' : 'white' }}
          />
        );
      }
      table.push(<Table.Row key={y} children={children} />);
    }

    return table;
  };

  return (
    <Table celled collapsing>
      <Table.Body>{createTable()}</Table.Body>
    </Table>
  );
};
