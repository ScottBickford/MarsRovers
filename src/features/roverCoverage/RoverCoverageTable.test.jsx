import React from 'react';
import renderer from 'react-test-renderer';
import { roverMovementsTestObject } from '../../App.test';
import { RoverCoverageTable } from './RoverCoverageTable';

describe('RoverInstructionsOutput', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<RoverCoverageTable upperRightCoords={roverMovementsTestObject.upperRightCoords} positions={roverMovementsTestObject.rovers[0].positions} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});