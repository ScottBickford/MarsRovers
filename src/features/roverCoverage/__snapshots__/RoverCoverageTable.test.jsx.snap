// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoverInstructionsOutput snapshot renders 1`] = `
<table
  className="ui celled collapsing table"
>
  <tbody
    className=""
  >
    <tr
      className=""
    >
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </tr>
    <tr
      className=""
    >
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </tr>
    <tr
      className=""
    >
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "lightblue",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </tr>
    <tr
      className=""
    >
      <td
        className=""
        style={
          Object {
            "backgroundColor": "lightblue",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "lightblue",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </tr>
    <tr
      className=""
    >
      <td
        className=""
        style={
          Object {
            "backgroundColor": "lightblue",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "lightblue",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </tr>
    <tr
      className=""
    >
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <td
        className=""
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </tr>
  </tbody>
</table>
`;
